src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -62,6 +62,10 @@
   (fn [request]
     (let [environment (get-environment-from-request request)]
       (cond
+        (and (nil? environment)
+             (= "puppet-v3-environment_modules-/*/" (get-in request [:route-info :route-id])))
+        (handler request)
```

Comment:
Might be better to have an overload of `wrap-with-environment-validation` which takes in an extra parameter that can be used to determine whether or not an environment is required.  That parameter could default to `false` to support other users of this endpoint for now but be set as `true` when called from the `environment-module-handler`.  The `cond` could then just thread through to the `handler` if no environment is provided, else throw the "environment parameter must be specified" error.  This way, we're not baking any special knowledge about endpoints into this implementation but allowing users of the function to vary the behavior, as desired.  WDYT?